net-core/api/src/main/java/org/pragmatica/net/http/HttpError.java

```diff
@@ -0,0 +1,66 @@
+package org.pragmatica.net.http;
```

Comment:
HttpError should be reworked. The idea is to create sealed interface with nested sealed interfaces grouping HTTP codes naturally (2xx codes, 3xx codes, 4xx codes and 5xx codes). Each group is represented by record. Top level interface adds status text method (use `text()` name) and status code (use `code()` name). The code() returns value from dedicated enum, not integer code. Top level interface implemend default `message()` method, which returns message from error code enum. No `body` and no `source` - HTTP errors are returned by external entity, we have no access to the cause behind them. Records which implement specific error codes contain `text` field, but for `code()`: this method is implemented explicitly and returns corresponding enum constant.